osu.Game/Screens/Select/Carousel/TopGlobalRank.cs

```diff
@@ -68,23 +73,53 @@ protected override void LoadComplete()
                     localScoresChanged);
             }, true);
 
-            void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet? changes)
+            void localScoresChanged(IRealmCollection<ScoreInfo> sender, ChangeSet changes)
             {
                 // This subscription may fire from changes to linked beatmaps, which we don't care about.
                 // It's currently not possible for a score to be modified after insertion, so we can safely ignore callbacks with only modifications.
                 if (changes?.HasCollectionChanges() == false)
                     return;
 
-                ScoreInfo? topScore = sender.MaxBy(info => (info.TotalScore, -info.Date.UtcDateTime.Ticks));
-                updateable.Rank = topScore?.Rank;
-                updateable.Alpha = topScore != null ? 1 : 0;
+                updateRank(sender);
+            }
+
+            updateRank();
+
+            void updateRank(IRealmCollection<ScoreInfo> sender = null)
+            {
+                getScoresRequest?.Cancel();
+                getScoresRequest = null;
+
+                getScoresRequest = new GetScoresRequest(beatmapInfo, beatmapInfo.Ruleset);
```

Comment:
A web request every time a new *local* score is added? 

This is not acceptable. The way this should be implemented is described in https://github.com/ppy/osu/discussions/26825.

Reply:
I'm afraid that the implementation requires forced saving of the best replays for each map selected by the user.